src/frontend/components/DashboardVideoLive/index.tsx

```diff
@@ -104,6 +131,54 @@ export const DashboardVideoLive = ({ video }: DashboardVideoLiveProps) => {
   const isLiveStarted =
     video.live_state !== undefined && video.live_state !== liveState.IDLE;
 
+  const lastArraySize = useRef(0);
+  const lastToastId = useRef<Nullable<string>>(null);
+
+  useEffect(() => {
+    if (
+      isLiveStarted &&
+      video.participants_asking_to_join.length !== 0 &&
+      video.join_mode !== JoinMode.FORCED &&
+      lastArraySize.current <= video.participants_asking_to_join.length &&
+      !(isPanelVisible && currentItem === LivePanelItem.VIEWERS_LIST)
+    ) {
+      if (lastToastId.current) {
+        toast.remove(lastToastId.current);
+      }
+      lastToastId.current = toast.custom(
+        <Box
+          background="bg-info"
+          direction="column"
+          gap="small"
+          round="xsmall"
+          pad="medium"
+        >
+          <Text color="blue-active" size="0.875rem">
+            {generateOnStageRequestNotificationMessage(
+              video.participants_asking_to_join,
+              intl,
+            )}
+          </Text>
+
+          <Button
+            primary
+            label={intl.formatMessage(messages.manageRequestBtnLabel)}
+            onClick={() => {
+              setPanelVisibility(true, LivePanelItem.VIEWERS_LIST);
+              toast.dismiss(lastToastId.current!);
+              lastToastId.current = null;
+            }}
+            size="small"
+          />
+        </Box>,
+        {
+          duration: Infinity,
+        },
+      );
+    }
+    lastArraySize.current = video.participants_asking_to_join.length;
+  }, [video.participants_asking_to_join]);
```

Comment:
dependencies are missing. At least all used in the first `if` statement.

Reply:
IMO it's correct, because in this precise case I only want my effect to be called when the video.participants_asking_to_join changed and not when any other parameters is changing (otherwise when the notification is closed but instructor leaves the viewers tab, a notification re-appears)

Reply:
IMO i only need this dependency here. With this dependency alone, the notification is updated only when it needed. If I add other dependency, the notification reappears when the instructor leaves th viewers tab.
But I know my way of address my needs is not the way recommanded by React (you are supposed to put in the dependency array all the variable / functions subject to change), so maybe I can find an other way. What do you think @RVassili ?